os/win/dnd.cpp

```diff
@@ -54,129 +54,139 @@ gfx::Point drag_position(HWND hwnd, POINTL& pt)
 
 // HGLOBAL Locking/Unlocking wrapper
 template<typename T>
-class GLock {
+class GLocked {
 public:
-  GLock() = delete;
-  GLock(const GLock&) = delete;
-  GLock(HGLOBAL hglobal) : m_hmem(hglobal) {
+  GLocked() = delete;
+  GLocked(const GLocked&) = delete;
+  GLocked(HGLOBAL hglobal) : m_hmem(hglobal) {
     m_data = static_cast<T>(GlobalLock(m_hmem));
   }
 
-  ~GLock() { GlobalUnlock(m_hmem); }
+  ~GLocked() { GlobalUnlock(m_hmem); }
 
   operator HGLOBAL() { return m_hmem; }
 
   operator T() { return m_data; }
 
+  bool operator==(std::nullptr_t) const { return m_data == nullptr; }
+  bool operator!=(std::nullptr_t) const { return m_data != nullptr; }
+
   SIZE_T size() { return GlobalSize(m_hmem); }
 
 private:
   HGLOBAL m_hmem;
   T m_data;
 };
 
+class DataWrapper {
+public:
+  DataWrapper() = delete;
+  DataWrapper(const DataWrapper&) = delete;
+  DataWrapper(IDataObject* data) : m_data(data) {}
+
+  ~DataWrapper() { release(); }
+
+  template<typename T>
+  GLocked<T> get(CLIPFORMAT cfmt) {
+    release();
+
+    FORMATETC fmt;
+    fmt.cfFormat = cfmt;
+    fmt.ptd = nullptr;
+    fmt.dwAspect = DVASPECT_CONTENT;
+    fmt.lindex = -1;
+    fmt.tymed = TYMED::TYMED_HGLOBAL;
+    m_result = m_data->GetData(&fmt, &m_medium);
+    if (m_result != S_OK)
+      return nullptr;
+
+    return GLocked<T>(m_medium.hGlobal);
+  }
+
+private:
+  void release() {
+    if (m_result == S_OK) {
+      ReleaseStgMedium(&m_medium);
```

Comment:
I have to fix this...I think that when the GLocked is destroyed, after it unlocks the HGLOBAL, then that is the moment to release the medium.

Reply:
OK. Just in case, I'd go back to the previous name `GLock` or renamed it to `HGlobalLock` (generally lock wrappers don't have a "locked" in its name, e.g. scoped_lock, lock_guard, etc,).

Reply:
Just as an aside: it actually made (and makes) sense to me that the thing returned is actually a locked HGLOBAL, and not a lock that I can lock or unlock. Anyway I'll return to the GLock. Thanks.

Reply:
I'd highly recommend to keep consistence of naming things with current standards or other project, you can search for example in other projects (e.g. [Chrome](https://source.chromium.org/chromium/chromium/src/+/main:base/win/scoped_hglobal.h;l=19?q=hglobal%20lock&ss=chromium)).